test/upload_validation_test.ts

```diff
@@ -4659,4 +4660,116 @@ describe('Pack metadata Validation', async () => {
       ]);
     });
   });
+
+  describe('Specific entity tables validate required properties are present', () => {
+    it('Succeeds when all required fields are present for users', async () => {
+      const syncTable = makeSyncTable({
+        name: 'Users',
+        identityName: 'User',
+        role: TableRole.Users,
+        schema: makeObjectSchema({
+          idProperty: 'id',
+          displayProperty: 'email',
+          userEmailProperty: 'email',
+          userIdProperty: 'id',
+          properties: {
+            id: {type: ValueType.String},
+            email: {type: ValueType.String, codaType: ValueHintType.Email},
+          },
+        }),
+        formula: {
+          name: 'Users',
+          description: '',
+          async execute([], _context) {
+            return {result: []};
+          },
+          parameters: [],
+          examples: [],
+        },
+      });
+      const metadata = createFakePackVersionMetadata(
+        compilePackMetadata({
+          version: '1',
+          syncTables: [syncTable],
+          defaultAuthentication: {
+            type: AuthenticationType.None,
+          },
+        }),
+      );
+      await doValidateJson(metadata);
+    });
+
+    it('Fail when required fields are missing for users', async () => {
```

Comment:
I think you didn't test the situation where the Users role is specified but the schema doesn't specify both `userEmailProperty` and `userIdProperty`... (and I think that test will fail if I'm reading your validation code correctly)

Reply:
Writing the test but we are validating`Error: Sync table schemas with role users must set a userEmailProperty`

Reply:
It fails in makeSyncTable (since that is where I put the validation) not in upload_validation. Can add an error there as well